libs/vkd3d/resource.c

```diff
@@ -875,6 +879,14 @@ HRESULT vkd3d_get_image_allocation_info(struct d3d12_device *device,
         allocation_info->Alignment = target_alignment;
     }
 
+    /* We can't know at this point whether the image is going to be CPU-visible
+     * or not, so always make sure we reserve enough space to store the linear
+     * subresource data as well. */
+    linear_size = vkd3d_compute_resource_layouts_from_desc(device, desc, NULL);
```

Comment:
I suppose the only plausible scenario for linear_size to be > optimal size is for D24S8? I suppose since we have to split the planes we end up with 32-bit + 8-bit?

Reply:
depth-stencil isn't actually allowed by the API here.

We shouldn't *really* be hitting this in practice with any of the formats we support, it's more a safeguard in case the driver is doing something really weird behind our backs. Can remove if you want.

Reply:
Is separate depth/stencil allowed?